gh-94628: Add explicit parameter list to sqlite3.connect docs (#94629)
python/cpython: Doc/library/sqlite3.rst

       or :const:`None` to disable opening transactions implicitly.
       See :ref:`sqlite3-controlling-transactions` for more.
   :type isolation_level: str | None

   :param check_same_thread:
       If :const:`True` (default), only the creating thread may use the connection.
       If :const:`False`, the connection may be shared across multiple threads;
       if so, write operations should be serialized by the user to avoid data
       corruption.
   :type check_same_thread: bool

   :param factory:
       A custom subclass of :class:`Connection` to create the connection with,
       if not the default :class:`Connection` class.
   :type factory: :class:`Connection`

   :param cached_statements:
       The number of statements that ``sqlite3``
       should internally cache for this connection, to avoid parsing overhead.
       By default, 128 statements.
   :type cached_statements: int

   :param uri:
       If set to :const:`True`, *database* is interpreted as a
       :abbr:`URI (Uniform Resource Identifier)` with a file path
       and an optional query string.
       The scheme part *must* be ``"file:"``,
       and the path can be relative or absolute.

>        The query string allows passing parameters to SQLite.
>        See :ref:`sqlite3-uri-tricks`.

Reviewer: ```suggestion
       The query string allows passing parameters to SQLite,
       enabling various :ref:`sqlite3-uri-tricks`.
```

A bit less choppy and better flow.

   :type uri: bool

>    :rtype: typing.Any

Reviewer: ```suggestion
   :rtype: sqlite3.Connection
```

Doesn't it return a `Connection` object (or a subclass thereof, if `factory` is passed)? This seems to be a lot more useful and less potentially confusing than resorting to a typing construct, no?

   https://www.sqlite.org/lang_transaction.html#deferred_immediate_and_exclusive_transactions

.. _sqlite3-uri-tricks:

SQLite URI tricks
-----------------

> Some useful URI tricks include::
> 
>    # Open a database in read-only mode.
>    con = sqlite3.connect("file:template.db?mode=ro", uri=True)
> 
>    # Don't implicitly create a new database file if it does not already exist.
>    # Will raise sqlite3.OperationalError if unable to open a database file.
>    con = sqlite3.connect("file:nosuchdb.db?mode=rw", uri=True)
> 
>    # Create a shared named in-memory database.
>    con1 = sqlite3.connect("file:mem1?mode=memory&cache=shared", uri=True)
>    con2 = sqlite3.connect("file:mem1?mode=memory&cache=shared", uri=True)
>    con1.execute("create table t(t)")
>    con1.execute("insert into t values(28)")
>    con1.commit()
>    rows = con2.execute("select * from t").fetchall()

Reviewer: ```suggestion
Some useful URI tricks include:

* Open a database in read-only mode::

      con = sqlite3.connect("file:template.db?mode=ro", uri=True)

  * Don't implicitly create a new database file if it does not already exist;
    will raise :exc:`~sqlite3.OperationalError` if unable to open one::
  
        con = sqlite3.connect("file:nosuchdb.db?mode=rw", uri=True)

  * Create a shared named in-memory database::
  
      con1 = sqlite3.connect("file:mem1?mode=memory&cache=shared", uri=True)
      con2 = sqlite3.connect("file:mem1?mode=memory&cache=shared", uri=True)
      con1.execute("create table t(t)")
      con1.execute("insert into t values(28)")
      con1.commit()
      rows = con2.execute("select * from t").fetchall()
```

Instead of keeping everything in one big verbatim code block, what about breaking this out into a bulleted list with each specific subtask, since each is self-contained and serves a different specific purpose?
